### PR TITLE
Swagger for telemetry APIs

### LINKED ISSUE #1093: Telemetry : Update swagger for metrics API
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
In V0.6.2 release of hotpot , Metric API requests do not execute as the controller enabled swagger validation. metrics APIs do not have swaager reference now.
**What you expected to happen**:
All metrics APIs shuld work as in Capri release. Need to have swaager for metrics APIS
**How to reproduce it (as minimally and precisely as possible)**:
clone v0.6.2 tag , install opensds with telemtetry. go to dashboard/telemetry . Tab to open Grafana and Alert Manager will not come up.

**Anything else we need to know?**:
NA
**Environment**:
- Hotpot(release/branch) version: v0.6.2
- OS (e.g. from /etc/os-release): Ubuntu
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by NajmudheenCT**
**What this PR does / why we need it**:
Adding Swagger for Telemetry APIS to Fix   https://github.com/opensds/opensds/issues/1093 .

**Which issue this PR fixes** 
 fixes #1093 
**Special notes for your reviewer**:
Below are the Metric APIS
/metrics GET/POST
/metrics/urls GET
/metrics/downloadConf GET
/metrics/uploadConf POST
**Test Report**
[telemetry_swagger_test.zip](https://github.com/opensds/opensds/files/3917032/telemetry_swagger_test.zip)

**Release note**:

```release-note
```

### CODE REVIEW

File: openapi-spec/swagger.yaml [hunk 1]
```diff
@@ -3003,3 +3117,80 @@ components:
...
+      description: >-
+        Collect Metric spec specifies the driver to trigger collect metrics action.
+      allOf:
+        - $ref: '#/components/schemas/BaseModel'
+        - type: object
+          required:
+            - driverType
+          properties:
+            driverType:
+              type: string
```
Comment (wisererik) [inline openapi-spec/swagger.yaml]: which driver types does it support? maybe enum is better here

File: openapi-spec/swagger.yaml [hunk 2]
```diff
@@ -3003,3 +3117,80 @@ components:
...
+              default: lvm
+    UrlSpec:
+      description: >-
+        Lists information for all url which can be used to access telemetry related services like grafana and alert manager.
+      type: object
+      required:
+        - name
+        - url
+        - desc
+      properties:
```
[the remark this answers is not in the document]
Comment (NajmudheenCT) [reply openapi-spec/swagger.yaml]: desc required by dashboard to display description for the URL. Everything supplied from backend.

File: openapi-spec/swagger.yaml [hunk 3]
```diff
@@ -1789,6 +1789,120 @@ paths:
...
+  '/v1beta/{tenantId}/metrics/uploadconf':
+    parameters:
+      - $ref: '#/components/parameters/tenantId'
+    post:
+      tags:
+        - Metrics
+      description: upload a config file.
+      parameters:
+      - name: conftype
+        in: header
```
[the remark this answers is not in the document]
Comment (NajmudheenCT) [reply openapi-spec/swagger.yaml]: conftype is required. I made required=true and enum provided for the allwed values.

File: openapi-spec/swagger.yaml [hunk 4]
```diff
@@ -3003,3 +3117,80 @@ components:
...
+              additionalProperties:
+                type: string
+              example:
+                key1: value1
+                key2: value2
+    CollectMetricSpec:
+      description: >-
+        Collect Metric spec specifies the driver to trigger collect metrics action.
+      allOf:
+        - $ref: '#/components/schemas/BaseModel'
```
[the remark this answers is not in the document]
Comment (NajmudheenCT) [reply openapi-spec/swagger.yaml]: Fixed, Thanks

File: openapi-spec/swagger.yaml (hunk 1 above)
Comment (NajmudheenCT) [reply openapi-spec/swagger.yaml]: lvm, ceph and huawei_oceanstor. enum provided for this.

**Comment by wisererik** [comment]
@NajmudheenCT please rebase and squash your commits

File: openapi-spec/swagger.yaml [hunk 5]
```diff
@@ -3158,3 +3282,82 @@ components:
                   - nvmeof
 
       description: Host is a consumer of volume from storage.
+    MetricSpec:
+      description: >-
+        Metric spec represents metrics of diifent type, which has name, instanceID label, timestamp , value etc..
```
Comment (kumarashit) [inline openapi-spec/swagger.yaml]:
```suggestion
        Metric spec represents metrics of different type, which has name, instanceID label, timestamp , value etc..
```

File: openapi-spec/swagger.yaml [hunk 6]
```diff
@@ -3158,3 +3282,82 @@ components:
...
+                type: object
+                properties:
+                  timestamp:
+                    type: integer
+                    format: int64
+                    description: Timestamp of the metric
+                    example: 1575363783
+                  value:
+                    type: number
+                    description: Vlue of the metric for this timestamp
```
Comment (kumarashit) [inline openapi-spec/swagger.yaml]:
```suggestion
                    description: Value of the metric for this timestamp
```

Comment (NajmudheenCT) [reply openapi-spec/swagger.yaml]: Thanks, Corrected

File: openapi-spec/swagger.yaml (hunk 5 above)
Comment (NajmudheenCT) [reply openapi-spec/swagger.yaml]: Thanks, Corrected

**Pull request merged**